woocommerce/woocommerce-ios
# [Mobile Payments] Tap to Pay reconnection analytics

Part of: #9735


## Description


We've added automatic reconnection to the Tap to Pay reader for eligible stores and devices.

This PR adds a `connection_type` property to relevant card reader connection analytics events, and sets it to `user_initiated` for existing flows, and `automatic_reconnection` for the new reconnection events.

## Testing instructions


1. Launch the app and take a payment using Tap to Pay on iPhone
2. Background the app
3. Foreground the app
4. Wait a few seconds
5. Observe that the `card_reader_connection_success` event is logged with the `connection_type: automatic_reconnection` property
6. Take a payment using a Bluetooth reader
7. Observe that after connection, the `card_reader_connection_success` event is logged with the `connection_type: user_initiated` property

## Screenshots

<img width="966" alt="`card_reader_connection_success` event console log with the `connection_type: automatic_reconnection` property" src="https://github.com/woocommerce/woocommerce-ios/assets/2472348/18b84a56-b1a1-41f1-a3c5-1eed075d3ed6">
<img width="948" alt="`card_reader_connection_success` and `card_reader_automatic_disconnect` event console logs with the `connection_type: user_initiated` property" src="https://github.com/woocommerce/woocommerce-ios/assets/2472348/bbe95f70-2017-46e5-8da9-71c6071abbb8">


---
